### ISSUE TITLE
Implement NN frontend functions in Ivy Experimental API

### ISSUE THREAD

**Issue by AnnaTz** [unifyai/ivy#11504]
**Torch frontend:**

pooling_functions:
- [ ] adaptive_avg_pool1d
- [ ] adaptive_avg_pool2d

loss_functions:
- [ ] cosine_embedding_loss
- [ ] smooth_l1_loss 
- [ ] huber_loss 
- [ ] l1_loss  
- [ ] nll_loss 
- [ ] soft_margin_loss 
- [ ] kl_div 
- [ ] margin_ranking_loss 
- [ ] poisson_nll_loss 
- [ ] hinge_embedding_loss 

non_linear_activation_functions:
- [ ] logsigmoid
- [ ] threshold
- [ ] elu
- [ ] celu
- [ ] mish
- [ ] selu
- [ ] prelu
- [ ] rrelu
- [ ] hardshrink
- [ ] softsign
- [ ] softshrink
- [ ] silu
- [ ] glu
- [ ] tanhshrink
- [ ] hardswish
- [ ] hardsigmoid
- [ ] hardtanh
- [ ] normalize
- [ ] group_norm

**Tensorflow frontend:**

**Jax frontend:**

-

> Please keep in mind that there might be overlap between this ToDo list and this https://github.com/unifyai/ivy/issues/3856. When taking on a function, please make sure to reserve it there as well.

-

**Comment by karalleyna** [comment]
Hi @AnnaTz,

I have implemented all activation functions in jax.nn.module [here](https://github.com/unifyai/ivy/pull/12502). Could you take a look at it?

**Comment by HadeerArafa** [comment]
- [ ] #23066

**Comment by samthakur587** [comment]
- [ ] sigmoid #23190

**Comment by dhanush-2501** [comment]
- [ ]#23255 
@AnnaTz I would like to work on this, this is the first i am trying an experimental API, can you pls provide some suggestions and resource to learn ?

**Comment by Vismay-dev** [comment]
[ ] #22996

**Comment by HadeerArafa** [comment]
- [ ] #23626

**Comment by HadeerArafa** [comment]
- [ ] #23660

**Comment by samthakur587** [comment]
- [ ] #23686

**Comment by samthakur587** [comment]
- [ ] #23688

**Comment by samthakur587** [comment]
- [ ] #23689

**Comment by SohamDas04** [comment]
normalize #24335

**Comment by OmerElshrief** [comment]
-[ ] #25942   (https://github.com/unifyai/ivy/issues/25942)

**Comment by samthakur587** [comment]
Hi! @AnnaTz is the ```rrelu``` non linear activation refers to   ```relu```  or not. and the ```threshold``` is refers to ```step threshold```?

**Comment by OmerElshrief** [comment]
(https://github.com/unifyai/ivy/issues/25942 )

**Comment by samthakur587** [comment]
- [ ]  #26093

**Comment by samthakur587** [comment]
- [ ]  #26094

**Comment by samthakur587** [comment]
- [ ]  #26095

**Comment by samthakur587** [comment]
- [ ]  #26096

**Comment by anudeeps0306** [comment]
#26101

**Comment by anudeeps0306** [comment]
#26102

**Comment by anudeeps0306** [comment]
#26103

**Comment by anudeeps0306** [comment]
#26104

**Comment by hnhparitosh** [comment]
- [ ] #26530

**Comment by Sujal-Burad** [comment]
-[ ] #26101

**Comment by Sujal-Burad** [comment]
- [ ] #26101

**Comment by Paarth002** [comment]
- [ ] #26715

**Comment by Sujal-Burad** [comment]
@AnnaTz , can I start working on this isssue?

> * [ ]  [l1_loss #26101](https://github.com/unifyai/ivy/issues/26101)

**Comment by leebissessar5** [comment]
- [ ] #26741

**Comment by hnhparitosh** [comment]
- [ ] #27154

**Comment by pilot-j** [comment]
#27434

**Comment by akshatvishu** [comment]
- [ ] #27784

**Comment by samthakur587** [comment]
- [ ]  #27816

**Comment by samthakur587** [comment]
- [ ] #27818
